### ISSUE TITLE
[FEATURE] No iOS Shortcut that allows interactive editing if given content

### ISSUE THREAD

**Issue by jmeosbn** [mastodon/mastodon-ios#651]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

If I use the mastodon shortcut and provide it with, say, some highlighted text from an article, it just posts it right away without allowing any additional text or edits.

### Expected Behavior

Compared with other (Twitter) clients, I have made the following shortcut, which would normally format the clipping I’ve highlighted, append the url, and provide a space above for me to insert any commentary / snark / etc.

![2EBC028A-C6D3-4C7A-9B63-1CE7074559A2](https://user-images.githubusercontent.com/530912/202896915-0fba1331-d765-447a-9436-fccbbf4684c8.jpeg)

However, while the app has a full featured post editor that can be run from the share sheet, this editor does not seem to be available via a shortcut so that I can preset text within it.

### Steps To Reproduce

Feed something to the only mastodon shortcut action listed.

Post is made without displaying an editor to amend or add to post.

I tried this with “Show when run” in case that made a difference.

### Environment

```markdown
- Device: iPhone 12
- OS: iOS
- Version: 16.1
- Build:
```

**Comment by jmeosbn** [comment]
Note that I'm currently using my previous Twitter shortcut to prepare posts for mastodon as follows. 😵‍💫

https://user-images.githubusercontent.com/530912/202901782-92e2a329-b12e-4b36-8b2d-fda5a0bb1dd9.MOV

**Comment by j-f1** [comment]
As a workaround you could use “Ask for Text” with a default answer and then pass that into the Mastodon action.

**Comment by jmeosbn** [comment]
Yeah, I didn't think that far as it wouldn't allow use of any editor features such as images etc.

**Comment by jmeosbn** [comment]
I did find a semi-workaround. I can use the standard “Share with App” action to open the standard post share extension.

However, the Shortcuts app doesn’t list Mastodon as an app I can pre-select, requiring me to choose the app each time!?!

![99CF3FE6-A5DC-42FE-982C-D6EE386D3B67](https://user-images.githubusercontent.com/530912/203097348-78816f7f-d241-406b-9308-225cd34350f0.jpeg)

Also, you lose the ability to change accounts for the post on the fly due to #666.

**Comment by MainasuK** [comment]
Please try using the share action then go back to check this list. The app is displayed on my side. And the "Post via Mastodon" Siri shortcut is designed to work silently. The image attachment is a TBD feature.

<img width=300 src="https://user-images.githubusercontent.com/7940186/203223757-ba7e9db9-5b9b-479e-b334-752ca8f960f1.PNG" />

**Comment by jmeosbn** [comment]
Yes, the silent action is not such a problem if I could get the shortcut to work.

Thanks for the suggestion (which I've just tried again to make sure) but that list just stays the same for me. 🤷🏻‍♂️

I've tried sharing to mastodon directly; I've tried via running the shortcut action; I've tried rebooting after.

I've updated to the latest iOS beta and it's still the same.
